core/src/main/java/org/apache/brooklyn/core/location/access/PublicNetworkFaceEnricher.java

```diff
@@ -0,0 +1,373 @@
+ * For example, to configure each Tomcat node to publish its mapped uri, and to use that sensor
+ * in Nginx for the target servers:
+ * <pre>
+ * {@code
+ * services:
+ * - type: cluster
+ *   id: cluster
+ *   brooklyn.config:
+ *    memberSpec:
+ *      $brooklyn:entitySpec:
+ *        type: org.apache.brooklyn.entity.webapp.tomcat.TomcatServer
+ *        brooklyn.enrichers:
+ *        - type: org.apache.brooklyn.core.location.access.PublicNetworkFaceEnricher
+ *          brooklyn.config:
+ *            sensor: main.uri
+ * - type: org.apache.brooklyn.entity.proxy.nginx.NginxController
+ *   brooklyn.config:
+ *     member.sensor.hostandport: $brooklyn:sensor("main.uri.mapped.public")
+ *     serverPool: cluster
+ * }
+ * </pre>
+ */
+@Beta
+public class PublicNetworkFaceEnricher extends AbstractEnricher {
+
+    // TODO Is this the best package for the enricher?
+    //
+    // TODO Need more logging, particularly for when the value has *not* been transformed.
+    //
+    // TODO What if the sensor has an unrelated hostname - we will currently still transform this!
+    // That seems acceptable: if the user configures it to look at the sensor, then we can make
+    // assumptions that the sensor's value will need translated.
+    //
+    // TODO If there is no port-mapping, should we advertise the original sensor value?
+    // That would allow the enricher to be used for an entity in a private network, and for
+    // it to be a no-op in a public cloud (so the same blueprint can be used in both). 
+    // However I don't think we should publish the original value: it could be the association
+    // just hasn't been created yet. If we publish the wrong (i.e. untransformed) value, that
+    // will cause other entity's using attributeWhenReady to immediately trigger.
+
+    private static final Logger LOG = LoggerFactory.getLogger(PublicNetworkFaceEnricher.class);
+
+    @SuppressWarnings("serial")
+    public static final ConfigKey<AttributeSensor<?>> SENSOR = ConfigKeys.newConfigKey(
+            new TypeToken<AttributeSensor<?>>() {}, 
+            "sensor",
+            "The sensor whose mapped value is to be re-published (with suffix \"mapped.public\"); "
+                    + "either 'sensor' or 'sensors' should be specified");
+
+    @SuppressWarnings("serial")
+    public static ConfigKey<Collection<? extends AttributeSensor<?>>> SENSORS = ConfigKeys.newConfigKey(
+            new TypeToken<Collection<? extends AttributeSensor<?>>>() {}, 
+            "sensors",
+            "The multiple sensors whose mapped values are to be re-published (with suffix \"mapped.public\"); "
+                    + "either 'sensor' or 'sensors' should be specified");
+
+    public static final ConfigKey<PortForwardManager> PORT_FORWARD_MANAGER = ConfigKeys.newConfigKey(
+            PortForwardManager.class, 
+            "portForwardManager",
+            "The PortForwardManager storing the port-mappings; if null, the global instance will be used");
```

Comment:
Yes, I think we will be deprecating this method (i.e. the non-default PFM, or maybe even the PFM) in light of the "Working with multipe networks" proposal.

A PFM just needs to be scoped such that there will not be clashes in the "publicIpId". That field came about from the early days of doing cloudstack integration. It's not really appropriate in many cases. In advanced networking use-cases, we've created a PFM for the "subnet" (which generally meant for the app, or a group of apps that we were carefully including as children of a single SubnetTier entity.

For now, I think we should assume "always using the default". I'm tempted to remove that as a config key from the enricher. But given it's a config key on other things, then it seemed sensible to do the same here. So I'll just leave it as-is for now. We can deprecate config/use of the PFM across the entire code-base if we decide to do that.